lib/mixlib/versioning/format/rubygems.rb

```diff
@@ -41,23 +45,56 @@ class Format
       #
       # @author Seth Chisamore (<schisamo@chef.io>)
       # @author Christopher Maier (<cm@chef.io>)
+      # @author Ryan Hass (<rhass@chef.io>)
       class Rubygems < Format
-        RUBYGEMS_REGEX = /^(\d+)\.(\d+)\.(\d+)(?:\.([[:alnum:]]+(?:\.[[:alnum:]]+)?))?(?:\-(\d+))?$/
-
         # @see Format#parse
         def parse(version_string)
-          match = version_string.match(RUBYGEMS_REGEX) rescue nil
+          # Pure regex along will never match partial versions correctly in the
+          # expected groupings. This is becausel partial versions shift the
+          # grouping values and break the mappings to for each appropriate value.
+          rubygems_version = version_string.scan(/[0-9]+|[a-z]+/i).map do |s|
+            /^\d+$/ =~ s ? s.to_i : s
+          end rescue nil
+
+          # Each value for MAJOR.MINOR.PATCH must be a integer.
+          unless rubygems_version && rubygems_version[0..2].map {|v| v.is_a?(Integer) }.all?
```

Comment:
`unless rubygems_version` is a no op when rubygems_version is an empty array . I don't think that's what we want here. Maybe? Unit test coverage should verify that.

This would only happen the method was passed nil, empty string or something we don't really expect (which I am not sure if that's already been handled before getting to this point).

Just a personal preference, I've also found using `unless` blocks with multiple conditions to use more brain power than simply making each condition contained.